From 9a5f8294a6e3f5315eb214873cc9c903c882f887 Mon Sep 17 00:00:00 2001
From: kszaq <kszaquitto@gmail.com>
Date: Fri, 10 Feb 2017 16:31:57 +0100
Subject: [PATCH] drivers/amlogic/display/vout: when switching frame rates
 during a video make sure to pick correct fine_tune_mode

Kodi switches frame rate after opening decoder. We need to adjust fine_tune_mode for requested frame rate according to FPS of video that is playing.
---
 drivers/amlogic/display/vout/tv_vout.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/drivers/amlogic/display/vout/tv_vout.c b/drivers/amlogic/display/vout/tv_vout.c
index 16ee9b8..3a1fee2 100644
--- a/drivers/amlogic/display/vout/tv_vout.c
+++ b/drivers/amlogic/display/vout/tv_vout.c
@@ -84,6 +84,11 @@ static struct vinfo_s *update_tv_info_duration(
 	enum vmode_e target_vmode, enum fine_tune_mode_e fine_tune_mode);
 static int hdmitx_is_vmode_supported_process(char *mode_name);
 
+static enum fine_tune_mode_e get_fine_tune_mode(
+	enum vmode_e mode, int fr_vsource);
+
+static int cur_fr_vsource = 0;
+
 #endif
 
 static int tv_vdac_power_level;
@@ -758,6 +763,9 @@ static int tv_set_current_vmode(enum vmode_e mode)
 	vout_log_info("%s[%d]fps_target_mode=%d\n",
 		      __func__, __LINE__, tvmode);
 
+	if (cur_fr_vsource != 0)
+		fine_tune_mode = get_fine_tune_mode(tvmode, cur_fr_vsource);
+
 	info->vinfo = update_tv_info_duration(tvmode, fine_tune_mode);
 #else
 	info->vinfo = get_tv_info(tvmode);
@@ -1084,6 +1092,7 @@ static int framerate_automation_process(int duration)
 		return 1;
 	}
 	fr_vsource = get_vsource_fps(duration);
+	cur_fr_vsource = fr_vsource;
 	fps_playing_flag = 0;
 	if ((fr_vsource == 5994)
 		|| (fr_vsource == 2997)
